<launch>
    <arg name="with_ppl_map" default="false" />
    <arg name="map" default=""/>
    <arg name="positions" default="/people_tracker/positions" />
    <arg name="pose" default="/people_tracker/pose" />
    <arg name="pose_array" default="/people_tracker/pose_array" />
    <arg name="people" default="/people_tracker/people" />

    <arg name="filtered_positions" default="/people_tracker_filter/positions" />
    <arg name="filtered_pose" default="/people_tracker_filter/pose" />
    <arg name="filtered_pose_array" default="/people_tracker_filter/pose_array" />
    <arg name="filtered_people" default="/people_tracker_filter/people" />
    <arg name="filtered_marker" default="/people_tracker_filter/marker_array" />

    <arg name="machine" default="localhost" />
    <arg name="user" default="" />

    <machine name="$(arg machine)" address="$(arg machine)" env-loader="$(optenv ROS_ENV_LOADER )" user="$(arg user)" default="true"/>

    <group if="$(arg with_ppl_map)">
        <node name="ppl_map_server" pkg="map_server" type="map_server" args="$(arg map)">
            <remap from="/map" to="/ppl_filter_map"/>
        </node>
    </group>

    <node name="people_tracker_filter" pkg="people_tracker_filter" type="people_tracker_filter" output="screen">
        <param name="map_topic" value="/ppl_filter_map" type="string" if="$(arg with_ppl_map)"/>
        <param name="map_topic" value="/map" type="string" unless="$(arg with_ppl_map)"/>
        <param name="positions" value="$(arg positions)" type="string"/>
        <param name="pose" value="$(arg pose)" type="string"/>
        <param name="pose_array" value="$(arg pose_array)" type="string"/>
        <param name="people" value="$(arg people)" type="string"/>
        <param name="filtered_positions" value="$(arg filtered_positions)" type="string"/>
        <param name="filtered_pose" value="$(arg filtered_pose)" type="string"/>
        <param name="filtered_pose_array" value="$(arg filtered_pose_array)" type="string"/>
        <param name="filtered_people" value="$(arg filtered_people)" type="string"/>
        <param name="filtered_marker" value="$(arg filtered_marker)" type="string"/>
    </node>

</launch>
